Mesh_3/examples/Mesh_3/draw_c3t3_surface.h

```diff
@@ -0,0 +1,265 @@
+    auto filter_end  = boost::make_filter_iterator(std::cref(filter), end, end);
+    auto it = std::min_element(filter_begin,
+                               filter_end,
+                               [&](auto v1, auto v2) {
+                                 return get(v_timestamp, v1) < get(v_timestamp, v2);
+                               });
+    if(it != filter_end) {
+      const auto min_timestamp_of_non_0_thread = *it;
+      timelineSlider->setMinimum(min_timestamp_of_non_0_thread);
+      timelineBox->setMinimum(min_timestamp_of_non_0_thread);
+      std::cerr << "Timeline: [ " << min_timestamp_of_non_0_thread
+                << " - " << timelineBox->maximum() << " ]\n";
+    }
+  }
+
+
+  QVariantAnimation rainbow_colors;
+
+  rainbow_colors.setKeyValueAt(0.0f, QVariant{ QColor::fromRgbF(0.75, 0.75, 1.  ) });
+  rainbow_colors.setKeyValueAt(0.2f, QVariant{ QColor::fromRgbF(0.  , 0.  , 1.  ) });
+  rainbow_colors.setKeyValueAt(0.4f, QVariant{ QColor::fromRgbF(0.  , 1.  , 0.  ) });
+  rainbow_colors.setKeyValueAt(0.6f, QVariant{ QColor::fromRgbF(1.  , 1.  , 0.  ) });
+  rainbow_colors.setKeyValueAt(0.8f, QVariant{ QColor::fromRgbF(1.  , 0.  , 0.  ) });
+  rainbow_colors.setKeyValueAt(1.0f, QVariant{ QColor::fromRgbF(0.5 , 0.  , 0.  ) });
+  rainbow_colors.setDuration(1000);
+
+  auto fromQColor = [](QColor c) -> CGAL::Color {
+    return {
+      (unsigned char)c.red(),
+      (unsigned char)c.green(),
+      (unsigned char)c.blue()
+    };
+  };
+
+  auto get_color = [&](double key) {
+    rainbow_colors.setCurrentTime(1000 * key);
+    return fromQColor( rainbow_colors.currentValue().value<QColor>() );
+  };
+
+  auto draw_mesh = [&]() {
+    for(auto fh : faces(mesh)) {
+      viewer->face_begin(CGAL::gray());
+      auto hd=mesh.halfedge(fh);
+      do
+        {
+          auto v = mesh.source(hd);
+          viewer->add_point_in_face(mesh.point(v), get(vnormals, v));
+          hd=mesh.next(hd);
+        }
+      while(hd!=mesh.halfedge(fh));
+      viewer->face_end();
+    }
+  };
+  bool call_redraw = false;
+  std::function<void()> recompute_elements = [&]() {
+    viewer->clear();
+    draw_mesh();
+    const auto thread_id_to_display = threadBox->value();
+    const auto max_timestamp = timelineSlider->value();
+    std::cerr << "Display thread #" << thread_id_to_display << '\n';
+    std::cerr << "Max timestamp: " << max_timestamp << '\n';
+    auto min = threadBox->minimum();
+    auto delta = threadBox->maximum()-threadBox->minimum();
+    for(auto v : vertices(mesh)) {
+      const auto ts = get(v_timestamp, v);
+      const auto thread_id = get(v_thread_id, v);
+      if((thread_id_to_display == 0 || thread_id == thread_id_to_display) &&
+         ts < max_timestamp)
+      {
+        CGAL::Color c = get_color(((0.+thread_id) - min) / delta);
+        viewer->add_point(mesh.point(v), c);
+      }
+    }
+    if(call_redraw) viewer->redraw();
```

Comment:
@gdamiand That is why I needed to turn `Basic_viewer_qt::redraw()` into a public method, in #4876, instead of protected: here in that example I use the raw `Basic_viewer_qt` class, unchanged, and a function `recompute_elements` (actually lambda function) that uses the methods `clear()`, `add_point`, `face_begin`, `add_point_in_face`, `face_end`, and `redraw()`.